src/parse/proto/extensions/simple_extension_uri.rs

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Parsing of [proto::extensions::SimpleExtensionUri].
+
+use thiserror::Error;
+use url::Url;
+
+use crate::{
+    parse::{context::ContextError, Anchor, Context, Parse},
+    proto,
+};
+
+/// A parsed [proto::extensions::SimpleExtensionUri].
+#[derive(Clone, Debug, PartialEq)]
+pub struct SimpleExtensionUri {
```

Comment:
A builder is going to need to do registration (not that you need that now).

Reply:
What do you mean with registration? The approach taken here is to have the parse context resolve, fetch and parse the referred extension (when parsing the definition):

https://github.com/substrait-io/substrait-rs/blob/dc945cd83aba8b55b90644c6f1f6e10dc5106621/src/parse/proto/extensions/simple_extension_uri.rs#L70-L72

Reply:
If an instance of this is created through the course of creating a plan (not the parse case you're concentrating on at the moment) the registration of anchors/reference ids would be handy.  It may happen in another class that generates these (perhaps providing function lookup and SQL to Substrait lookup).  I will be fleshing out a builder in substrait-python soon. 
 A builder (and registration) isn't a concern for this PR.